### ISSUE TITLE
Dataloader and batch size

### ISSUE THREAD

**Issue by enrico310786** [CRCTransformers/deepdive-book#3]
Hi,

in the notebook Chapter-2-Machine-Translation.ipynb you define the train, valid and test dataloader as

train_iter = DataLoader(train_data, batch_size=BATCH_SIZE, shuffle=True, collate_fn=generate_batch)

with BATCH_SIZE= 16 and generate_batch a custom function. Now, if i iterate over the dataloader, i should observe 16 pairs of sentences per iteration, instead here i see just one sentence per iteration. Thus i do not use the entire dataset to train and validate the model. If I'm wrong, where am I wrong?

**Comment by enrico310786** [comment]
Hi, i found my error. Sorry!
